src/storage/src/source/kafka.rs

```diff
@@ -1100,8 +1113,15 @@ impl KafkaSourceReader {
         } else {
             *last_offset_ref = offset_as_i64;
 
-            let ts = Partitioned::new_singleton(RangeBound::exact(partition), offset);
-            Some((message, ts, 1))
+            let output = message.map(|msg| SourceOutput {
+                key: msg.key,
+                value: msg.value,
+                metadata: msg.metadata,
+                from_time: Partitioned::new_singleton(RangeBound::exact(partition), offset),
+            });
+            let emit_offset = MzOffset::from(std::cmp::max(offset.offset, *partition_as_of));
```

Comment:
Here we are performing the fast forward action

Reply:
```suggestion
            // Fast Forwarding the offset we emit at to the max offset in the partition when it was discovered
            let emit_offset = MzOffset::from(std::cmp::max(offset.offset, *partition_as_of));
```